### ISSUE TITLE
[Bug]: Querynode panic runtime error: invalid memory address or nil pointer dereference when normal search

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#22289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230218-57f8de95
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.3.0.dev38
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy cluster with 4 querynodes
2. connect and set resource group like this:
```
Describe resource group:__default_resource_group, ResourceGroupInfo:
        <name:__default_resource_group>,
        <capacity:1000000>,
        <num_available_node:0>,
        <num_loaded_replica:{}>,
        <num_outgoing_node:{}>,
        <num_incoming_node:{}> (base.py:436)
[2023-02-20 04:55:09,509 -  INFO - fouram]: [Base] Describe resource group:RG_0, ResourceGroupInfo:
        <name:RG_0>,
        <capacity:2>,
        <num_available_node:2>,
        <num_loaded_replica:{'fouram_Za6yprCT': 1}>,
        <num_outgoing_node:{}>,
        <num_incoming_node:{}> (base.py:436)
[2023-02-20 04:55:09,510 -  INFO - fouram]: [Base] Describe resource group:RG_1, ResourceGroupInfo:
        <name:RG_1>,
        <capacity:2>,
        <num_available_node:2>,
        <num_loaded_replica:{'fouram_Za6yprCT': 1}>,
        <num_outgoing_node:{}>,
        <num_incoming_node:{}>
```
4. create a collection
```
Collection schema: {'auto_id': False, 'description': '', 'fields': [{'name': 'id', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float_vec
tor', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}, {'name': 'int64_1', 'description': '', 'type': <DataType.INT64: 5>}, {'name': 'int64_2', 'description': '', 'type': <DataType.INT64: 5>}, {'name': 'floa
t_1', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'double_1', 'description': '', 'type': <DataType.DOUBLE: 11>}, {'name': 'varchar_1', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 256}}]}
```
5. Build index `{'index_type': 'IVF_FLAT', 'metric_type': 'L2', 'params': {'nlist': 512}}`
6. Insert 2m entities with ni=5w
7. Load 2 replicas into 2 resource groups: `[RG_0, RG_1]`
and replica info:
```
collection fouram_Za6yprCT replicas info: Replica groups:
- Group: <group_id:439581042011013121>, <group_nodes:(2, 1)>, <shards:[Shard: <channel_name:fouram-scale-55qfb-gmlmd-op-9-4964-rootcoord-dml_1_439581040684564484v1>, <shard_leader:1>, <shard_nodes:[1, 1, 1, 2, 2]>, Shard: <channel_name:four
am-scale-55qfb-gmlmd-op-9-4964-rootcoord-dml_0_439581040684564484v0>, <shard_leader:2>, <shard_nodes:[2, 1, 1, 2, 2]>]>
- Group: <group_id:439581042011013122>, <group_nodes:(6, 7)>, <shards:[Shard: <channel_name:fouram-scale-55qfb-gmlmd-op-9-4964-rootcoord-dml_1_439581040684564484v1>, <shard_leader:7>, <shard_nodes:[7, 7, 7, 6, 6]>, Shard: <channel_name:four
am-scale-55qfb-gmlmd-op-9-4964-rootcoord-dml_0_439581040684564484v0>, <shard_leader:6>, <shard_nodes:[6, 7, 7, 6, 6]>]>
```
8. Search successfully for a while and one querynode panic:
```
[2023/02/20 04:55:22.932 +00:00] [DEBUG] [querynode/service.go:94] [QueryNode] [port=21123]
[2023/02/20 04:55:22.932 +00:00] [DEBUG] [querynode/service.go:111] ["QueryNode connect to etcd successfully"]
[2023/02/20 04:55:22.932 +00:00] [INFO] [config/refresher.go:57] ["start refreshing configurations"] [source=EtcdSource]
[2023/02/20 04:55:22.932 +00:00] [INFO] [management/server.go:68] ["management listen"] [addr=:9091]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x29e3527]

goroutine 385 [running]:
panic({0x335ddc0, 0x4e5bf00})
    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc0005ef040 sp=0xc0005eef80 pc=0x11edcf7
runtime.panicmem(...)
    /usr/local/go/src/runtime/panic.go:220
runtime.sigpanic()
    /usr/local/go/src/runtime/signal_unix.go:818 +0x336 fp=0xc0005ef090 sp=0xc0005ef040 pc=0x1204d16
github.com/milvus-io/milvus/internal/querynode.(*QueryNode).GetDataDistribution(0xc000d05600, {0x354d0c0?, 0x361b420?}, 0xc000472360)
    /go/src/github.com/milvus-io/milvus/internal/querynode/impl.go:1299 +0x67 fp=0xc0005ef4e8 sp=0xc0005ef090 pc=0x29e3527
github.com/milvus-io/milvus/internal/distributed/querynode.(*Server).GetDataDistribution(0x0?, {0x3add658?, 0xc000473dd0?}, 0x0?)
    /go/src/github.com/milvus-io/milvus/internal/distributed/querynode/service.go:323 +0x2b fp=0xc0005ef518 sp=0xc0005ef4e8 pc=0x2bbf04b
github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_GetDataDistribution_Handler.func1({0x3add658, 0xc000473dd0}, {0x353bce0?, 0xc000472360})
    /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5811 +0x78 fp=0xc0005ef558 sp=0xc0005ef518 pc=0x24922d8
github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor({0x3add658?, 0xc000473dd0?}, {0x353bce0, 0xc000472360}, 0xc0005ef5b8?, 0xc00052fd88)
    /go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22 +0x49 fp=0xc0005ef588 sp=0xc0005ef558 pc=0x2475da9
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3add658?, 0xc000473dd0?}, {0x353bce0?, 0xc000472360?})
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc0005ef5c8 sp=0xc0005ef588 pc=0x1c5a23a
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3add658, 0xc000472330}, {0x353bce0, 0xc000472360}, 0xc0005b7e20, 0xc0005b7e60)
    /go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.37.0/interceptor.go:349 +0x528 fp=0xc0005ef938 sp=0xc0005ef5c8 pc=0x19787a8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3add658?, 0xc000472330?}, {0x353bce0?, 0xc000472360?})
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a fp=0xc0005ef978 sp=0xc0005ef938 pc=0x1c5a23a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x3add658, 0xc000472330}, {0x353bce0, 0xc000472360}, 0xc000116a18?, 0x335c980?)
    /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf fp=0xc0005ef9d0 sp=0xc0005ef978 pc=0x1c5a0df
github.com/milvus-io/milvus/internal/proto/querypb._QueryNode_GetDataDistribution_Handler({0x361b420?, 0xc0008ddc20}, {0x3add658, 0xc000472330}, 0xc0006e2460, 0xc00041f110)
    /go/src/github.com/milvus-io/milvus/internal/proto/querypb/query_coord.pb.go:5813 +0x138 fp=0xc0005efa28 sp=0xc0005ef9d0 pc=0x2492198
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000879c20, {0x3aedbc8, 0xc000d116c0}, 0xc00021a480, 0xc00041f290, 0x4e6fd40, 0x0) 
    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13 fp=0xc0005efe48 sp=0xc0005efa28 pc=0x175ed73
google.golang.org/grpc.(*Server).handleStream(0xc000879c20, {0x3aedbc8, 0xc000d116c0}, 0xc00021a480, 0x0) 
    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b fp=0xc0005eff68 sp=0xc0005efe48 pc=0x1763d7b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x98 fp=0xc0005effe0 sp=0xc0005eff68 pc=0x175c7d8
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc0005effe8 sp=0xc0005effe0 pc=0x12230a1
created by google.golang.org/grpc.(*Server).serveStreams.func1
    /go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x28a
```

### Steps To Reproduce

```markdown
argo-workflow: fouram-scale-55qfb-gmlmd
```

### Milvus Log

devops cluster, chaos-testing ns pods:
```
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-datacoord-747c95m2vbq   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-datanode-5bf4c65x2nr9   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-indexcoord-85b7bc2qml   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-indexnode-7cc9c9r7z7l   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-proxy-78fbcb895dzgqt2   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querycoord-84949fv7v7   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querynode-6b4cc46dqtf   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querynode-6b4cc4kchcs   1/1     Running     2 (19m ago)    25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querynode-6b4cc4xhrp2   1/1     Running     0              25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querynode-6b4cc4z86rg   1/1     Running     1 (19m ago)    25m
fouram-scale-55qfb-gmlmd-op-9-4964-milvus-rootcoord-75b7cctlxzp   1/1     Running     0              25m
```
pod `fouram-scale-55qfb-gmlmd-op-9-4964-milvus-querynode-6b4cc4kchcs` previous log:
[qn1_pre.log](https://github.com/milvus-io/milvus/files/10779831/qn1_pre.log)

**Comment by yah01** [comment]
/assign @ThreadDao 
fixed with #22290

**Comment by ThreadDao** [comment]
I did not reproduce this problem on image `master-20230221-c3e8ad36`. Close it first. open it if reproduce
